include/gauge_field_order.h

```diff
@@ -945,4 +945,29 @@ namespace quda {
   };
 
 
-}
+  
+  // Use traits to reduce the template explosion
+  template<typename ,QudaReconstructType,int N=18> struct gauge_mapper { };
```

Comment:
Seems like you never stop learning C++ 😉. Makes sense though but can't remember having seen that before.